### PR TITLE
Use UMD (with tests) + Add ES6 transpiling

### PR DESCRIPTION
**Pull request by g-plane**
What I have done:
- Use UMD to bundle this library. So this library can be used in browser and node.js
- Add browsers test. (Using Karma)
- Use babel to transpile ES6 code. (to ES5)
- Simplify code. (Using arrow function mainly)

### CODE REVIEW

File: src/methods/nth.js [hunk 1]
```diff
@@ -1,11 +1,9 @@
 'use strict';
 
 module.exports = function nth(n, offset) {
-  const _offset = offset || 0;
+  offset = offset || 0;
```
Comment (BrainMaestro) [inline src/methods/nth.js]: Since we're transpiling, we could just use default arguments

Comment (g-plane) [inline src/methods/nth.js]: Yeah, I forget it. Thanks for you reminding!

**Comment by ecrmnn** [comment]
Awesome work @g-plane 

Could you please check your PR with the latest release?

**Comment by g-plane** [comment]
Anything else should be changed?

**Comment by ecrmnn** [comment]
Thanks again @g-plane, but could you submit UMD-testing as its own PR?

This got a little overwhelming, and I feel like the ES5 transpiling have been taken care of already. I don't think we need to change that. Also I don't like shorthand conditionals, so we can skip those changes too.

**Comment by g-plane** [comment]
So keeping using Webpack but not change the code?

**Comment by ecrmnn** [comment]
Do we need webpack for testing UMD?

**Comment by g-plane** [comment]
Webpack is just a tool to bundle. But before the test, we should generate the single file library.

**Comment by g-plane** [comment]
If you don't want to change the code style in this PR, I can rebase it. But we should use Webpack with babel-loader to make a UMD module, instead of using babel-cli.

**Comment by g-plane** [comment]
Should I make a change?

**Comment by ecrmnn** [comment]
Ok. Skip the code style changes but use webpack. 👍

**Comment by g-plane** [comment]
Are there any problems?

**Comment by ecrmnn** [comment]
No problems @g-plane, just havn't got around to take another look at it. Hopefully I'll get it merged tonight 😄

**Comment by g-plane** [comment]
Thanks.

**Comment by ecrmnn** [comment]
Hm. It's failing for me

```bash
28 06 2017 19:24:56.537:INFO [karma]: Karma v1.7.0 server started at http://0.0.0.0:9876/
28 06 2017 19:24:56.540:INFO [launcher]: Launching browser Chrome with unlimited concurrency
28 06 2017 19:24:56.545:INFO [launcher]: Starting browser Chrome
28 06 2017 19:24:57.342:INFO [Chrome 58.0.3029 (Mac OS X 10.11.6)]: Connected on socket P5b4ArppkMo_JsD2AAAA with id 81333593
Chrome 58.0.3029 (Mac OS X 10.11.6): Executed 0 of 0 ERROR (0.002 secs / 0 secs)
npm ERR! Test failed.  See above for more details.
```

**Comment by g-plane** [comment]
Have you installed all dependencies and pulled all the code? It seemed that there were no problems both on my Windows and macOS.
image

**Comment by ecrmnn** [comment]
Yup. Everything installed.

**Comment by g-plane** [comment]
Sorry that I can't reproduce it. How about try cleaning and removing `node_modules` directory then running `npm install` to re-install?
The unit test and other configuration files may be OK. As you can see, it worked on CI.

**Comment by ecrmnn** [comment]
Yes I've removed and reinstalled all modules.
I can not understand why it's not passing for me.

```bash
01 07 2017 13:49:33.280:INFO [karma]: Karma v1.7.0 server started at http://0.0.0.0:9876/
01 07 2017 13:49:33.283:INFO [launcher]: Launching browser Chrome with unlimited concurrency
01 07 2017 13:49:33.289:INFO [launcher]: Starting browser Chrome
01 07 2017 13:50:33.291:WARN [launcher]: Chrome have not captured in 60000 ms, killing.
01 07 2017 13:50:35.296:WARN [launcher]: Chrome was not killed in 2000 ms, sending SIGKILL.
01 07 2017 13:50:37.298:WARN [launcher]: Chrome was not killed by SIGKILL in 2000 ms, continuing.
npm ERR! Test failed.  See above for more details.
```

**Comment by g-plane** [comment]
Really confusing. Could you do more tests on other OSs or machines?
According to the given log, it seems failed to connect Karma with Chrome.

**Pull request closed without merging**